client/query/query.go

```diff
@@ -24,12 +24,24 @@ func getStatusFromChannel(c *channel.Channel) ChannelStatus {
 		}
 		return Closing
 	}
+
 	if !c.PostFundComplete() {
 		return Proposed
 	}
 	return Ready
 }
 
+func getVirtualChannelStatus(c *channel.Channel, virtualAppDef types.Address) ChannelStatus {
+	// ADR 0009: Allows for intermediaries to exit the protocol before receiving all signed post funds
+	isVirtual := c.AppDefinition == virtualAppDef
```

Comment:
Is it a bit awkward to call the function `getVirtualChannelStatus` and then test if the channel isVirtual inside the function body?

I also note that we have a `VirtualChannel` type, which we could use to differentiate / handle this as a special case. 

I think overall I am a bit wary of having a different status reported magically for intermediaries, when the underlying channel state is quite different. On my alternative #1272, I addressed this by introducing more fine-grained statuses to use.

Reply:
Having thought about it some more, I'm coming around to the idea that the query status can deviate from the "true channel status".

Reply:
I do think we want to tidy up the fn though, seems quite messy to have to thread `virtualAppDef` through everywhere. If we could use something like a type assertion / type check instead that might make it much cleaner.

Reply:
Actually I think we could just remove this check for now. Nowhere else in `GetPaymentChannelInfo` do we ensure that the channel is a virtual channel. So it's strange to do a specific check when getting a virtual channel status.

I do think we should have a check to make sure the channel is the right type, but that can be tackled outside of this PR.

Reply:
I've updated this in https://github.com/statechannels/go-nitro/pull/1271/commits/32bf07346cb1d4e3621220cfde4067afefd6faf4 so we no longer check the `AppDef` or pass around the virtual app address.

I've also created #1280.